vault/src/robocorp/vault/__init__.py

```diff
@@ -79,91 +16,91 @@ def _get_vault():
     return Vault()
 
 
-def get_secret(secret_name: str) -> SecretContainer:
-    """Get secret defined with given name.
+def get_secret(name: str, hide: bool = True) -> SecretContainer:
+    """Get a secret with the given name.
 
     Args:
-        secret_name: Name of secret to fetch.
+        name: Name of secret to fetch
+        hide: Hide secret values from log output
 
     Note:
-        The returned secret is not cached, so, calling this function again
-        may do a new network roundtrip.
-
-    Note:
-        When used, if `robocorp.log` is in the environment, all the values
-        gotten will be automatically hidden from the logs.
-        i.e.: For each value, `log.hide_from_output(value)` will be called.
+        If `robocorp.log` is available in the environment, the `hide` argument
+        controls if the given values are automatically hidden in the log
+        output.
 
     Returns:
         The container for the secret (which has key-value pairs).
 
     Raises:
         RobocorpVaultError: Error with API request or response payload.
     """
-    try:
-        from robocorp import log  # type: ignore [attr-defined]
-
-        ctx = log.suppress_variables()
-        has_log = True
-    except ImportError:
-        pass  # If robocorp.log is not being used that's OK.
-        ctx = nullcontext()
-        has_log = False
-
-    with ctx:
+    with _suppress_variables():
         vault = _get_vault()
+        secret = vault.get_secret(name)
 
-        # Note: secrets are always gotten from the backend when requested
-        # (so any updates will be gotten in a new `get_secret` call).
-        secret = vault.get_secret(secret_name)
-
-        if has_log:
-            # There's no caching in place, so, on any new call it's possible
-            # that values changed, so, call this every time.
-            _ignore_secret_values(secret)
+        if hide:
+            _hide_secret_values(secret)
 
-    return secret
+        return secret
 
 
-def set_secret(secret: SecretContainer) -> None:
-    """Overwrite an existing secret with new values.
+def set_secret(
+    name: str,
+    values: dict[str, Any],
```

Comment:
I had an `update_secret`, but I removed it just before making the PR. I think I'll add it back.

It could be kept backwards compatible, that's true, it's just unfortunate naming. I think a pair of getter and setter is the obvious API, but the setter needing some specific container instance is not that great.

Reply:
I think that if the `set_secret` was kept backward compatible while also supporting the new API (so, using `@overload` in typing) would be a reasonable compromise (it's a bit strange for me if the `get` returns something but the `set` can't deal with the same object returned).

Reply:
I stared at it a bit, and I think overloading the method made the functionality a bit too confusing. I went with the other option of adding a `create_secret` method and keeping things backwards compatible.